### ISSUE TITLE
Provide examples of diagnostics using executable

### ISSUE THREAD

**Issue by euoia** [jose-elias-alvarez/null-ls.nvim#127]
### Issues

- [X] I have checked [existing issues](https://github.com/jose-elias-alvarez/null-ls.nvim/issues) and there are no existing ones with the same request.

### Feature description

I'd be interested in helping contribute some additional builtins for some command line tools that I use. To help get started, it would be really helpful if an example was provided that shows a diagnostics executable being used, perhaps with an example of generator_factory. The current examples seems to work directly on the contents of the buffer rather than parsing the output of a command line tool. All of the real-world examples, of course, use the make_builtin helper, which has a different format. The documentation isn't really clear on whether it's okay to use make_builtin for extensions, or during testing.

Ideally, the documentation would include a complete example of how to create and register a new diagnostic helper that parses the output from a command-line executable. Even better, perhaps suggest a way to easily test the pattern matching.

Thanks again for this, by the way.

### Help

Yes, but I don't know how to start. I would need guidance

**Comment by jose-elias-alvarez** [comment]
Agreed that this would be a big improvement. The examples in the readme are really just supposed to show how null-ls itself works, but as you note they aren't relevant to built-ins. I'd be happy to accept a PR for a quick guide or can write one myself whenever I have time.

**Comment by jose-elias-alvarez** [comment]
I took a stab at improving the examples in 0c2eab0a7c7389b01b9c326bac30782fb5687e11. I want to keep improving the documentation in the future, but I think this is good enough for now.
